File: level5.py
# runlength = lambda t: t[0] * t[1]
# wholeline = lambda l: "".join(map(runlength, l))
# wholebanner = lambda b: "\n".join(map(wholeline, b))
# import pickle
# print wholebanner(pickle.load(file("level5_banner.p")))

import pickle
import json

with open("level5_banner.p") as in_file:
    with open("level5_banner.json", "w") as out_file:
        json.dump(pickle.load(in_file), out_file)

File: level5.go
package main

// pickle python specific, so let's just learn how to use JSON in Go

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"
)

func FileContent(name string) []byte {
	r, err := os.Open(name)
	if err != nil {
		panic(err)
	}
	defer r.Close()
	bytes, err := ioutil.ReadAll(r)
	if err != nil {
		panic(err)
	}
	return bytes
}

func main() {
	bytes := FileContent("level5_banner.json")

	var m [][][]interface{}
	json.Unmarshal(bytes, &m)

	for _, line := range m {
		for _, tuple := range line {
			how_many := int(tuple[1].(float64))
			char := tuple[0].(string)
			for i := 0; i < how_many; i++ {
				fmt.Print(char)
			}
		}
		fmt.Println("")
	}
}
